internal/resources/ekscluster/resource_ekscluster_test.go

```diff
@@ -0,0 +1,599 @@
+					Taints: []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolTaint{
+						{
+							Effect: eksmodel.NewVmwareTanzuManageV1alpha1EksclusterNodepoolTaintEffect(eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolTaintEffectPREFERNOSCHEDULE),
+							Key:    "randomkey",
+							Value:  "randomvalue",
+						},
+					},
+					InstanceTypes: []string{},
+				},
+			},
+		},
+	}
+}
+
+func bodyInspectingResponder(t *testing.T, expectedContent interface{}, successResponse int, successResponseBody interface{}) httpmock.Responder {
+	return func(r *http.Request) (*http.Response, error) {
+		successFunc := func() (*http.Response, error) {
+			return httpmock.NewJsonResponse(successResponse, successResponseBody)
+		}
+
+		if expectedContent == nil {
+			return successFunc()
+		}
+
+		// Compare to expected content.
+		expectedBytes, err := json.Marshal(expectedContent)
+		if err != nil {
+			t.Fail()
+			return nil, err
+		}
+
+		if r.Body == nil {
+			t.Fail()
+			return nil, fmt.Errorf("expected body on request")
+		}
+
+		bodyBytes, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Fail()
+			return nil, err
+		}
+
+		// Map of map of strings for comparing subnet equality
+		subnetMap := make(map[string]map[string][]string, 0)
+
+		var bodyInterface map[string]interface{}
+		if err = json.Unmarshal(bodyBytes, &bodyInterface); err == nil {
+			var expectedInterface map[string]interface{}
+
+			err = json.Unmarshal(expectedBytes, &expectedInterface)
+			if err != nil {
+				return nil, err
+			}
+
+			diff := deep.Equal(bodyInterface, expectedInterface)
+			if diff == nil {
+				return successFunc()
+			} else {
+				// special check for subnets
+				// First, populate all the diffs pertaining to subnets into maps
+				for _, diffItem := range diff {
+					isSubnetKey := strings.Contains(diffItem, "map[subnetIds]") && strings.Contains(diffItem, ".slice")
+					isExpectedTag := strings.Contains(diffItem, "map[subnetIds]") && strings.Contains(diffItem, ".slice")
+
+					if isSubnetKey {
+						segments := strings.Split(diffItem, ":")
+						key := strings.Split(segments[0], ".slice")[0]
+
+						// Create map if not present
+						if subnetMap[key] == nil {
+							subnetMap[key] = make(map[string][]string, 0)
+						}
+
+						// Add vals to map
+						vals := strings.Split(segments[1], "!=")
+						subnetMap[key]["left"] = append(subnetMap[key]["left"], strings.TrimSpace(vals[0]))
+						subnetMap[key]["right"] = append(subnetMap[key]["right"], strings.TrimSpace(vals[1]))
+					}
+
+					if !(isSubnetKey || isExpectedTag) {
+						t.Fail()
+						return nil, errors.New("diff identified outside of subnet order and additional VMware tags")
+					}
+				}
+
+				// Then, sort slices and compare
+				for _, set := range subnetMap {
+					left := set["left"]
+					right := set["right"]
+
+					// sort
+					sort.Strings(left)
+					sort.Strings(right)
+
+					subnetDiff := deep.Equal(left, right)
```

Comment:
can you use the deep.Equal from the testing/require package ? 
what is the advantage of this new deep package ?

Reply:
It returns a human readable diff to be compared- I will attempt to replace this tomorrow.